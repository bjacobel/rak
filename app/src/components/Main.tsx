import { useQuery } from '@tanstack/react-query';
import { styled } from '@linaria/react';

import { getData, Data } from '../services/data';
import ErrorComponent from './errors/ErrorComponent';
import { lightgrey } from '../stylesheets/colors.css';
import { center, openSans } from '../stylesheets/shared.css';
import StyledLink from './StyledLink';

const DataHeader = styled.h3`
  ${center}
  ${openSans}
  ${lightgrey}
  font-style: italic;
`;

const Logo = styled.div`
  height: 100px;
  background-image: url('../assets/images/logo.svg');
  background-repeat: no-repeat;
  background-position: center;
`;

export default () => {
  const { isLoading, data, error } = useQuery<Data, Error>({ queryKey: ['data'], queryFn: getData });

  if (error) {
    return <ErrorComponent error={error} />;
  }

  if (isLoading) {
    return <span>loading...</span>;
  }

  return (
    <>
      <Logo role="banner" />
      <DataHeader>{data!.text || ''}</DataHeader>
      <StyledLink to="/child/foo">Routing demo</StyledLink>
      <StyledLink to="/asdf">404 demo</StyledLink>
      <StyledLink to="/error">Error</StyledLink>
    </>
  );
};
