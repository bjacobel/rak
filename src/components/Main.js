import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

import styles from '../stylesheets/main.css';
import { link } from '../stylesheets/link.css';
import { getDataAsync } from '../actions/data';

const mapStateToProps = state => ({
  data: state.data,
});

const mapDispatchToProps = {
  getDataAsync,
};

class Main extends Component {
  componentDidMount() {
    this.props.getDataAsync();
  }

  render() {
    const { data } = this.props;

    return (
      <>
        <div className={styles.logo} />
        <h3 className={styles.data}>{data.text || ''}</h3>
        <Link className={link} to="/child/foo">
          Routing demo
        </Link>
        <Link className={link} to="/asdf">
          404 demo
        </Link>
      </>
    );
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Main);
