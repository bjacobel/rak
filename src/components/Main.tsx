import React from 'react';
import { Link } from 'react-router-dom';
import { useQuery } from '@tanstack/react-query';

import { getData, Data } from '../services/data';
import ErrorComponent from './errors/ErrorComponent';
import { data as dataStyle, logo } from '../stylesheets/main.css';
import { link } from '../stylesheets/link.css';

export default () => {
  const { isLoading, data, error } = useQuery<Data>(['data'], getData);

  if (error) {
    return <ErrorComponent error={error} />;
  }

  if (isLoading) {
    return null;
  }

  return (
    <>
      <div role="banner" className={logo} />
      <h3 className={dataStyle}>{data!.text || ''}</h3>
      <Link className={link} to="/child/foo">
        Routing demo
      </Link>
      <Link className={link} to="/asdf">
        404 demo
      </Link>
    </>
  );
};
