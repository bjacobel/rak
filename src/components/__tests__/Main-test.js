import React from 'react';
import { shallow } from 'enzyme';

import Main from 'components/Main';

describe('main component', () => {
  describe('un-Connected component', () => {
    it('matches snapshot', () => {
      expect(shallow(<Main.WrappedComponent data={{ text: 'foo' }} getDataAsync={jest.fn()} />)).toMatchSnapshot();
    });

    it("renders an h3 with data if it didn't hit an error", () => {
      const main = shallow(<Main.WrappedComponent data={{ text: 'foo' }} getDataAsync={jest.fn()} />);

      expect(main.find('h3').length).toBe(1);
      expect(main.find('h3').text()).toEqual('foo');
    });

    it('renders no data but an error if data-fetching hits an error', () => {
      const main = shallow(<Main.WrappedComponent data={{}} errors={['error']} getDataAsync={jest.fn()} />);

      expect(main.find('h3').length).toBe(1);
      expect(main.find('h3').text()).toEqual('');
    });

    it('calls getDataAsync on mount', () => {
      const getDataAsync = jest.fn();
      shallow(<Main.WrappedComponent data={{ text: 'foo' }} getDataAsync={getDataAsync} />);
      expect(getDataAsync).toHaveBeenCalled();
    });
  });
});
