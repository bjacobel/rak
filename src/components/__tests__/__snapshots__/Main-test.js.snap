// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main component un-Connected component matches snapshot 1`] = `
<Fragment>
  <div />
  <h3>
    foo
  </h3>
  <Link
    to="/child/foo"
  >
    Routing demo
  </Link>
  <Link
    to="/asdf"
  >
    404 demo
  </Link>
</Fragment>
`;
